pkg/yamlgen/yamlgen.go

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package yamlgen
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"go/types"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
+	"github.com/dave/jennifer/jen"
+	"github.com/pkg/errors"
+)
+
+// TODO(saswatamcode): Add tests.
+// TODO(saswatamcode): Check jennifer code for some safety.
+// TODO(saswatamcode): Add mechanism for caching output from generated code.
+// TODO(saswatamcode): Currently takes file names, need to make it module based(something such as https://golang.org/pkg/cmd/go/internal/list/).
+
+// GenGoCode generates Go code for yaml gen from structs in src file.
+func GenGoCode(src []byte) (string, error) {
```

Comment:
Good for start, but I wonder if we can scope to just one desired structure, to limit imports. Maybe it's later optimization - so fine for now

Reply:
Can we add todo for that?